src/Api/Api.php

```diff
@@ -23,31 +25,58 @@ class Api
 {
     private LoggerInterface $logger;
 
-    private Client $client;
-
     private Config $config;
 
-    public function __construct(LoggerInterface $logger, Client $client, Config $config)
+    private Client $client;
+
+    public function __construct(LoggerInterface $logger, Config $config, Client $client)
     {
         $this->logger = $logger;
-        $this->client = $client;
         $this->config = $config;
+        $this->client = $client;
+    }
+
+    /**
+     * Send request and load next pages, if "nextLink" is present in the response.
+     * Returns decoded JSON body
+     * @param Request $request
+     * @return Generator|array[]
+     */
+    public function send(Request $request): Generator
```

Comment:
Cele je to vlastne iba rozhranie na API, tu sa spustaju requesty a nacitavaju dalsie stranky.